exercises/allergies/test/Tests.hs

```diff
@@ -98,3 +129,81 @@ specs = do
                                           , Shellfish
                                           , Strawberries
                                           , Tomatoes     ]
+
+            -- Property: For an arbitrary `allergen` and its `score`,
+            -- `allergies score` is a list of exactly the element
+            -- `allergen`.
+
+            it "accepts single allergens" $ forAll allergenWithScore $
+              \(allergen, score) -> allergies score == [allergen]
+
+            -- Property: For an arbitrary `score` and arbitrary `allergen`
+            -- that does not match it, that `allergen` is not an element of
+            -- `allergies score`.
+
+            it "rejects mismatching allergens" $ forAll complementWithScore $
+              \(allergen, score) -> allergen `notElem` allergies score
+
+            -- Property: For an arbitrary set of `allergens` and their
+            -- combined `score`, `allergies score` is a list of exactly
+            -- `allergens`.
+
+            it "accepts multiple allergens" $
+              forAllShrink allergensWithScore shrinkAllergensWithScore $
+                \(allergens, score) -> allergies score == allergens
+
+            -- Property: For an arbitrary `score` and all `allergens` that
+            -- are not part of that score, there is no overlap between
+            -- `allergens` and `allergies score`.
+
+            it "rejects multiple mismatching allergens" $
+              forAllShrink complementsWithScore shrinkComplementsWithScore $
+                \(allergens, score) -> null (allergens `intersect` allergies score)
```

Comment:
for similar reasons as the `notElem` test, I think this test is not necessary, since exact list equality of `allergies` already means no unnecessary allergens can be in the result of `allergies`. So the above "accepts multiple allergens" would have to also use an `intersect`, or this test should be removed

Reply:
I agree to removing it.